slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/AggregateTest.scala

```diff
@@ -356,32 +357,44 @@ class AggregateTest extends AsyncTest[RelationalTestDB] {
       assertNesting(q7, 1)
     }
 
-    case class BData(id: String, a: String, b: String)
+    case class StrWrapper(v: String)
+    object StrWrapper {
+      def applyOption(v: Option[String]): Option[StrWrapper] = v.map(StrWrapper(_))
+      def unapplyOption(v: Option[StrWrapper]): Option[Option[String]] = Some(v.map(_.v))
+    }
+    
+    case class BData(id: String, a: String, b: String, c: Option[StrWrapper])
     class B(t: Tag) extends Table[BData](t, None, "B_DISTINCT") {
       def id = column[String]("id")
       def a = column[String]("a")
       def b = column[String]("b")
+      def c = column[Option[String]]("c")
+
+      def cWrapped = c <> (StrWrapper.applyOption, StrWrapper.unapplyOption)
 
-      def * = (id, a, b) <> ((BData.apply _).tupled, BData.unapply)
+      def * = (id, a, b, cWrapped) <> ((BData.apply _).tupled, BData.unapply)
     }
 
     val bs = TableQuery[B]
 
-    val datab = Set( BData("1", "a", "a")
-                   , BData("2", "a", "b")
-                   , BData("3", "c", "b")
+    val datab = Set( BData("1", "a", "a", Some(StrWrapper("v1")))
+                   , BData("2", "a", "b", None)
+                   , BData("3", "c", "b", Some(StrWrapper("v1")))
                    )
 
-    val qb1a = bs.map(_.a).distinct
+    val qb1a = bs.map(_.a).distinctOn(identity)
     val qb1b = bs.distinctOn(_.a).map(_.a)
     val qb2 = bs.distinctOn(_.a).map(a => (a.a, 5))
     val qb3a = bs.distinctOn(_.a).map(_.id).filter(_ === "1") unionAll bs.distinctOn(_.a).map(_.id).filter(_ === "2")
-    val qb4 = bs.map(a => (a.a, a.b)).distinct.map(_._1)
+    val qb4 = bs.map(a => (a.a, a.b)).distinctOn(a => (a._1, a._2)).map(_._1)
```

Comment:
Sorry, i don't understand difference with `.distinctOn(identity)` (possibly, my Slick skills is not enought here).

Reply:
just want to cover all variation, we shouldn't rely on function implementation in tests